File: src/flink/trading-stream-agg/src/main/java/com/exchange/job/common/OrderBookEventType.java
package com.exchange.job.common;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author uuhnaut69
 */
@Getter
@AllArgsConstructor
public enum OrderBookEventType {
    TRADING_RESULT("TRADING_RESULT");
    private final String value;
}

File: src/exchange/src/main/kotlin/com/exchange/exchange/domain/orderbook/OrderBookEventType.kt
package com.exchange.exchange.domain.orderbook

import com.exchange.exchange.core.EventType

/**
 * @author uuhnaut69
 *
 */
enum class OrderBookEventType(
    override val type: String
) : EventType {
    TRADING_RESULT("TRADING_RESULT"),
    CREATE_ASK_LIMIT_ORDER_SUCCESS("CREATE_ASK_LIMIT_ORDER_SUCCESS"),
    CREATE_ASK_LIMIT_ORDER_FAILED("CREATE_ASK_LIMIT_ORDER_FAILED"),
    CREATE_BID_LIMIT_ORDER_SUCCESS("CREATE_BID_LIMIT_ORDER_SUCCESS"),
    CREATE_BID_LIMIT_ORDER_FAILED("CREATE_BID_LIMIT_ORDER_FAILED"),
    CANCEL_ORDER_SUCCESS("CANCEL_ORDER_SUCCESS"),
    CANCEL_ORDER_FAILED("CANCEL_ORDER_FAILED"),
    ;
}
